### ISSUE TITLE
Reading permission on macOS

### ISSUE THREAD

**Issue by beijoca** [xournalpp/xournalpp#1757]
**Affects versions :**
 - OS: macOS 10.15.3 Catalina
 - Version of Xournal++: 1.0.17

**Describe the bug**
When:
-trying to save a Xournal .xopp file
-trying to open a pdf from within the application
-trying to save an annotated pdf file
reading permission isn't granted. 

**To Reproduce**

1) create a new file in Xournal
2) save as
3) select a folder

or:

1) open a pdf file in the Finder (Open with > Xournal++.app)
2) save (or save as)

or:
1) Annotate PDF

or:
1) Open

**Expected behavior**
to be able read folder contents

**Screenshots of Problem**
![Screenshot 2020-03-02 at 14 11 08](https://user-images.githubusercontent.com/14167966/75679572-2c823c00-5c90-11ea-8e92-dd3c6e37ff3f.png)

**Additional context**
It is possible to open files via the Finder (select file > open with > Xournal++app)
it is possible to open recent documents from within the application
it is possible to save a .xopp file (after ignoring the warning "could not read the contents of ...")

In the Security & Privacy panel, I gave Xournall++.app "Full Disk Access". This didn't help.

**Comment by beijoca** [comment]
Some more thoughts on this issue:

- at first I thought that this problem relates to the non-native file chooser used by Xournal++ on mac. (Because applications that use the native file chooser (e.g. MrWriter) bring up the macOS dialog where one is asked to grant permission access a certain folder)

- However, gimp 2.10 doesn't use the native file chooser. Nonetheless, the access dialog is triggered by gimp (see screenshot).

![Screenshot 2020-03-07 at 08 42 30](https://user-images.githubusercontent.com/14167966/76139701-14753880-6053-11ea-9c45-046de79fbe9c.png)

**Comment by beijoca** [comment]
the problem probably also extends to 10.14 macOS Mojave. I can't confirm myself, as I jumped from 10.13 High Sierra directly to Catalina. With Mojave, Apple introduced this safety feature were the user needs to manually grant permission to apps.

**Comment by thosoo** [comment]
Workaround: Running xournalpp as root.
`sudo /Applications/Xournal++.app/Contents/MacOS/xournalpp`

**Comment by Technius** [comment]
Similar issues have been reported for other programs such as Geany, Darktable, etc.

Can you try
* Starting Xournal++ from dock
* Manually locating and starting the Xournal++ executable
* Running from terminal

I don't have a MacOS device so I can only provide limited support.

**Comment by thosoo** [comment]
Starting Xournal++ from dock: Doesnt work
Manually locating and starting the Xournal++ executable: Doesnt work
Running from terminal: Seems working, even without sudo.

Following error came up after this:
![Screenshot 2020-03-21 20 25 46](https://user-images.githubusercontent.com/8445255/77234979-8a7cb200-6bb2-11ea-912d-ac5127498f93.png)

**Comment by beijoca** [comment]
thanks thosoo, that workaround is perfectly acceptable for now!

**Comment by fredowski** [comment]
Hi, is this maybe related to the same problem as in pspp:

https://savannah.gnu.org/bugs/?58128#comment2

Friedrich

**Comment by beijoca** [comment]
Reading/writing permissions in MacOS 10.15 can be set manually:

https://support.apple.com/guide/mac-help/control-access-files-folders-mac-mchld5a35146/mac
https://www.wikihow.com/Give-System-Permissions-for-Apps-on-MacOS-Catalina

It remains to be solved however, how MacOS can be triggered to ask for permission, whenever the Application is launched for the first time (or whenever it tries to read/write a file for the first time)

**Comment by fredowski** [comment]
If fixed the identical problem for the GNU pspp MacOS bundle with this commit:

http://git.savannah.gnu.org/cgit/pspp.git/commit/?id=69a18e796323f9c8796984a1d0a3532fc34aba50

You must not use the shell script in the bundle Xournal++.app/Contents/MacOS/xournalpp . The problem with that on Catalina is that the permissions are not valid for the final binary Xournal++.app/Contents/MacOS/xournalpp-bin . So you must call the binary in the bundle directly and not via a shell script. In order to set the environment variables you must do this insided the binary with code.

**Comment by whiterock** [comment]
> If fixed the identical problem for the GNU pspp MacOS bundle with this commit:
> 
> http://git.savannah.gnu.org/cgit/pspp.git/commit/?id=69a18e796323f9c8796984a1d0a3532fc34aba50
> 
> You must not use the shell script in the bundle Xournal++.app/Contents/MacOS/xournalpp . The problem with that on Catalina is that the permissions are not valid for the final binary Xournal++.app/Contents/MacOS/xournalpp-bin . So you must call the binary in the bundle directly and not via a shell script. In order to set the environment variables you must do this insided the binary with code.

What environment variables do I need to set so I can call xournalpp-bin? If I call it directly from zsh it gives me:
```dyld: Library not loaded: @rpath/libzip.5.dylib
  Referenced from: /Applications/Xournal++.app/Contents/MacOS/xournalpp-bin
  Reason: image not found
zsh: abort      /Applications/Xournal++.app/Contents/MacOS/xournalpp-bin
```

I can call xournalpp with sudo, the downside here though is, that all files created then are owned by root, which in and of itself might be okay, but files owned by root don't get synced by iCloud... and of course it should not be needed to use sudo to open this amazing program here :)

**Comment by fredowski** [comment]
> What environment variables do I need to set so I can call xournalpp-bin? If I call it directly from zsh it gives me:

I think you can see the necessary environment variables in the commit:

http://git.savannah.gnu.org/cgit/pspp.git/commit/?id=69a18e796323f9c8796984a1d0a3532fc34aba50

**Comment by Indomitable** [comment]
Hi, 
   Today I faced this issue what I found is that it is possible to create a new folder from Xournal save dialog and then you will have rights to save/load files there. 
  One more thing is that opening files from Finder doesn't work, but works from the application open dialog.

**Comment by beijoca** [comment]
> I have this issue on Big Sur 11.6. It is possible to save a file after ignoring the 'cannot read folder' error, but not to open a file from within the app. Also, pdf cannot be opened, it gives a dialogue: 'Xournal++ cannot open files in the “Apple PDF pasteboard type” format.'
> 
> This seems not a small issue to me. I'm just checking out the app and it is really not usable like this on a Mac.

I'm on MacOS 11.6 (on a MBP with Intel Core i5) and Xournal 1.0.19 works for me. I had problems with newer versions of Xournalpp as well, although I don't recognize the error you're reporting.

**Comment by opsnull** [comment]
yes。 1.0.19 is fine。The newest version can‘t open file。

**Comment by scateu** [comment]
+1

**Comment by pelafustan** [comment]
I'm on macOS Monterey 12.2.1, and neither 1.0.19 and latest stable release can access to my personal folders. Also tried with nightly build and still cannot access to my files.

**Comment by philippwulff** [comment]
I want to point out that this problem persists

**Comment by rolandlo** [comment]
> I want to point out that this problem still exists

I know. We need someone with an Apple Developer subscription for signing the binary, or the user will have to self-sign it, otherwise the permission issue will not be solved.

**Comment by sebastiankmilo** [comment]
error fixed in new version, i install version 1.1.2 and work.

**Comment by pelafustan** [comment]
> error fixed in new version, i install version 1.1.2 and work.

I can confirm that. I installed version 1.1.2 from homebrew and works as to be intended. Thanks!
